src/DocBlock/Annotation.php

```diff
@@ -16,10 +16,16 @@
  * This represents an entire annotation from a docblock.
  *
  * @author Graham Campbell <graham@alt-three.com>
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
  */
 class Annotation
 {
     /**
+     * @internal
+     */
+    const REGEX_TYPES = '(?<types>(?<type>(?:[\\\\@$]?\w+(?:\[\])?)|(?<generic>[\\\\@$]?\w+<(?:[\\\\@$]?\w+,\s*)?(?:(?&types)|(?&generic))>))(?:\|(?:(?&type)|(?&generic)))*)';
```

Comment:
astually, we have `@this` in `PhpdocReturnSelfReferenceFixerTest`.
i will add test to `AnnotationTest` as well